src/libstd/sys_common/thread_parker.rs

```diff
@@ -0,0 +1,80 @@
+// Copyright 2016 Amanieu d'Antras
+//
+// Licensed under the Apache License, Version 2.0, <LICENSE-APACHE or
+// http://apache.org/licenses/LICENSE-2.0> or the MIT license <LICENSE-MIT or
+// http://opensource.org/licenses/MIT>, at your option. This file may not be
+// copied, modified, or distributed except according to those terms.
+
+//! A simple spin lock based thread parker. Used on platforms without better
+//! parking facilities available.
+
+use sync::atomic::{AtomicBool, Ordering};
+use time::Instant;
+use sync::atomic::spin_loop_hint;
+
+// Helper type for putting a thread to sleep until some other thread wakes it up
+pub struct ThreadParker {
+    parked: AtomicBool,
+}
+
+impl ThreadParker {
+    pub const IS_CHEAP_TO_CONSTRUCT: bool = true;
+
+    pub fn new() -> ThreadParker {
+        ThreadParker {
```

Comment:
```suggestion
        Self {
```

There are a bunch of other places this applies in this PR.